libparsec/crates/client/tests/unit/client/update_user_profile.rs

```diff
@@ -0,0 +1,109 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use libparsec_client_connection::test_register_send_hook;
+use libparsec_protocol::authenticated_cmds;
+use libparsec_tests_fixtures::prelude::*;
+use libparsec_types::prelude::*;
+
+use crate::ClientUserUpdateProfileError;
+
+use super::utils::client_factory;
+
+#[parsec_test(testbed = "coolorg", with_server)]
+async fn ok(env: &TestbedEnv) {
```

Comment:
We should probably test all combinations for the `ok` case:
- Standard -> Admin
- Standard -> Outsider
- Admin -> Standard
- Admin -> Outsider
- Outsider -> Standard
- Outsider -> Admin

Also we could add another test `ok_idempotent` for:
- Admin -> Admin
- Standard -> Standard
- Outsider -> Outsider

Also we should add tests for the "self downgrading" case (either ok or not ok depending on what is decided):
- Admin (self) -> Standard
- Admin (self) -> Outsider